review docs + other consistency fixes (#414)
neicnordic/sensitive-data-archive: sda/cmd/finalize/finalize.md

### RabbitMQ broker settings

These settings control how `finalize` connects to the RabbitMQ message broker.

- `BROKER_HOST`: hostname of the RabbitMQ server
- `BROKER_PORT`: RabbitMQ broker port (commonly: `5671` with TLS and `5672` without)

> - `BROKER_QUEUE`: message queue to read messages from (commonly: `accessionIDs`)

Reviewer: ```suggestion
- `BROKER_QUEUE`: message queue to read from (commonly: `accession`)
```

> - `BROKER_ROUTINGKEY`: message queue to write success messages to (commonly: if `ARCHIVE_` and `BACKUP_` storage backend are set queue should be `completed` otherwise `backup`)

Reviewer: ```suggestion
- `BROKER_ROUTINGKEY`: Routing key for publishing messages (commonly: `completed`)
```

Messages is always sent to `completed` now since `finalize` performs archive backup if configured to do so.

- `*_PORT`: S3 connection port (default: `443`)
- `*_REGION`: S3 region (default: `us-east-1`)
- `*_CHUNKSIZE`: S3 chunk size for multipart uploads.
- `*_CACERT`: Certificate Authority (CA) certificate for the storage system, this is only needed if the S3 server has a certificate signed by a private entity

and if `*_TYPE` is `POSIX`:

- `*_LOCATION`: POSIX path to use as storage root

## Service Description

`Finalize` adds stable, shareable _Accession ID_'s to archive files.
If a backup location is configured it will perform backup of a file.

> When running, `finalize` reads messages from the configured RabbitMQ queue (commonly: `accessionIDs`).

Reviewer: ```suggestion
When running, `finalize` reads messages from the configured RabbitMQ queue (commonly: `accession`).
```

2. If the service is configured to perform backups i.e. the `ARCHIVE_` and `BACKUP_` storage backend are set. Archived files will be copied to the backup location.
   1. The file size on disk is requested from the storage system.
   2. The database file size is compared against the disk file size.
   3. A file reader is created for the archive storage file, and a file writer is created for the backup storage file.
3. The file data is copied from the archive file reader to the backup file writer.
4. If the type of the `DecryptedChecksums` field in the message is `sha256`, the value is stored.
5. A new RabbitMQ `complete` message is created and validated against the `ingestion-completion` schema. If the validation fails, an error message is written to the logs.
6. The file accession ID in the message is marked as *ready* in the database. On error the service sleeps for up to 5 minutes to allow for database recovery, after 5 minutes the message is Nacked, re-queued and an error message is written to the logs.
7. The complete message is sent to RabbitMQ. On error, a message is written to the logs.
8. The original RabbitMQ message is Ack'ed.

## Communication

> - `Finalize` reads messages from one RabbitMQ queue (commonly: `accessionIDs`).

Reviewer: ```suggestion
- `Finalize` reads messages from one RabbitMQ queue (commonly: `accession`).
```

Author: this is where I am bit confused because helm charts use `accessionIDs` queue

> - `Finalize` writes messages to one RabbitMQ queue (commonly: if `ARCHIVE_` and `BACKUP_` storage backend are set queue should be `completed` otherwise `backup`).

Reviewer: ```suggestion
- `Finalize` publishes messages with one routing key  (commonly: `completed`).
```
